### ISSUE TITLE
german chargepoint data

### ISSUE THREAD

**Issue by klaernie** [openchargemap/ocm-system#230]
From reading around in the repo, I noticed that the importer for data from the Bundesnetzagentur has been removed, but the data remains, same with the Bundesnetzagentur being listed as a data provider. If I understand correctly this has been triggered by the intention to only implement providers that support [OCPI](https://github.com/ocpi/ocpi).

From reading around in the german press, I can only find industry associations requesting the legislative branch to support OCPI, but no communication from the Bundesnetzagentur that they would like to support this or provide a proper API.

However one of the german states - Baden Württemberg - provides an open data portal providing the data:
https://mobidata-bw.de/dataset/e-ladesaulen

According to the description this seems to be OCPI compatible, would this a way forward to integrate the data?

Best regards,
Andre

**Comment by webprofusion-chrisc** [comment]
Thanks Andre, yes it looks like we could perhaps consume https://api.mobidata-bw.de/ocpdb/api/public/v1/locations

Is there a statement somewhere on that site for the data license?

**Comment by areleu** [comment]
The source he shared has a CC BY 4.0 notice on the left, and they use the same bnetza data which is also CC BY 4.0 https://www.bundesnetzagentur.de/DE/Fachthemen/ElektrizitaetundGas/E-Mobilitaet/Ladesaeulenkarte/start.html Not sure if they are in sync, since BNETZA does not offer an API they, probably download and update the data at some interval. Or probably they do not update it at all:

> Erstellt 8. Juli 2022, 13:20 (UTC+02:00)

The BNETZA has data from December 12 2024. 

I do a regular dump of the BNETZA data using this script: https://github.com/areleu/fair-charging-station-data and publish to Zenodo: https://zenodo.org/records/12683380 (these are normalised, I will probably dump also the raw csvfied versions) to have an archive from their older reports, otherwise these get lost. I could figure out serving it as an OCPI API if that is interesting to the project, let me know if you think doing such a thing would be valuable otherwise I will just keep doing the regular dumps (which itself is a pain because the agency keeps changing the format of their report).

**Comment by webprofusion-chrisc** [comment]
I think the problem we had with the original data was that each time we imported it all of the IDs for each location would change, I could be wrong but that's just my recollection. This caused problems in de-duplication for us so at the time it wasn't worth battling with. 

We would prefer to read data from as close to the source as possible, but if the source if outdated then it doesn't really benefit our dataset, so the question is can they be encouraged to keep their OCPI data feed up to date.

**Comment by areleu** [comment]
> I think the problem we had with the original data was that each time we imported it all of the IDs for each location would change, I could be wrong but that's just my recollection. This caused problems in de-duplication for us so at the time it wasn't worth battling with.

Fair enough, this is an issue that I also often face when dealing with that dataset. There is work being done on the interoperability of this kind of data in Germany. I will keep this issue in mind and see if I can reach someone from the BNETZA to encourage them to use the OCPI or at least provide something with more consistency across versions.

**Comment by klaernie** [comment]
If I recall correctly the portal by BW is using the implementation coined by https://github.com/binary-butterfly/ocpdb, but I've seen the data been refreshed at least twice in late December.

**Comment by webprofusion-chrisc** [comment]
Thanks, so I think the BW portal is a third party aggregator and is not making specific claims as to the data license and may or may not include sources which haven't authorized distribution under a particular license.

**Comment by klaernie** [comment]
I would consider this as their official claim (bordered in red):
![image](https://github.com/user-attachments/assets/52e6ca01-641b-46f3-81db-98dfa1c2e9a3)

They also repeat it in the "more information" link of the OCPDB-API: https://mobidata-bw.de/dataset/e-ladesaulen/resource/cec90b3e-19ad-4ca6-bd5d-900ce2ad23cd

While they technically are a third party, I would consider them a reliable third party, since this portal is a official offering by an entity considered part of the state. So in my personal interpretation they are the closest we can ever get to the data collected by the BNetzA, until the BNetzA manages to offer a proper API.

The current state of the data in openchargemap is from my personal perspective confusing - I find data for charging stations in Germany, but the data is definitely out of date. I can see that the BNetzA is listed as a provider, but also their map interfaces shows stations which are not contained in openchargemap. Would I be a novice user I would be immediately disappointed and would never consider openchargemap again, despite it being a reliable provider in other regions of the world.

**Comment by areleu** [comment]
> They also repeat it in the "more information" link of the OCPDB-API: [mobidata-bw.de/dataset/e-ladesaulen/resource/cec90b3e-19ad-4ca6-bd5d-900ce2ad23cd](https://mobidata-bw.de/dataset/e-ladesaulen/resource/cec90b3e-19ad-4ca6-bd5d-900ce2ad23cd)
> 
> While they technically are a third party, I would consider them a reliable third party, since this portal is a official offering by an entity considered part of the state. So in my personal interpretation they are the closest we can ever get to the data collected by the BNetzA, until the BNetzA manages to offer a proper API.

The OCPDB still has the problem of [deduplication ](https://github.com/binary-butterfly/ocpdb/blob/main/docs/matching.md), they do a a solid guess, but still a guess when it comes to identifying individual charging stations. This issue can only be solved by having the BNEtzA being more rigorous with these reports. This is not beyond their capabilities, I have seen they do a good job with the https://www.marktstammdatenregister.de/MaStR/, however, this is likely a different working group.
